src/transformers/models/llava_onevision/modeling_llava_onevision.py

```diff
@@ -0,0 +1,675 @@
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        vision_feature_layer = (
+            vision_feature_layer if vision_feature_layer is not None else self.config.vision_feature_layer
+        )
+        vision_feature_select_strategy = (
+            vision_feature_select_strategy
+            if vision_feature_select_strategy is not None
+            else self.config.vision_feature_select_strategy
+        )
+        vision_aspect_ratio = (
+            vision_aspect_ratio if vision_aspect_ratio is not None else self.config.vision_aspect_ratio
+        )
+
+        if (input_ids is None) ^ (inputs_embeds is not None):
+            raise ValueError(
+                "You cannot specify both input_ids and inputs_embeds at the same time, and must specify either one"
+            )
+
+        if (pixel_values is not None or pixel_values_videos is not None) and inputs_embeds is not None:
+            raise ValueError(
+                "You cannot specify both pixel_values/pixel_values_videos and inputs_embeds at the same time, and must specify either one"
+            )
+
+        if inputs_embeds is None:
+            inputs_embeds = self.get_input_embeddings()(input_ids)
+
+        for pixels, sizes, special_token in [
+            (pixel_values, image_sizes, self.config.image_token_index),
+            (pixel_values_videos, image_sizes_videos, self.config.video_token_index),
+        ]:
+            if pixels is None:
+                continue
+
+            # if videos are being processed, we can flatten frame dim and treat it as a huge batch of images
+            if pixels.dim() == 6:
+                sizes = [imsize for size_list in sizes for imsize in size_list]
+                batch_size, frames, num_patches, channels, height, width = pixels.shape
+                pixels = pixels.view(batch_size * frames, num_patches, channels, height, width)
+
+            image_num_patches = [
+                image_size_to_num_patches(
+                    image_size=imsize,
+                    grid_pinpoints=self.config.image_grid_pinpoints,
+                    patch_size=self.config.vision_config.image_size,
+                )
+                for imsize in sizes
+            ]
+
+            # figure out if pixel_values is concatenated or stacked
+            if pixels.dim() == 5:
+                _pixel_values_list = [pix_val[:num_patch] for pix_val, num_patch in zip(pixels, image_num_patches)]
+                # [batch_size*frames*num_patches, num_channels, height, width]
+                pixels = torch.cat(_pixel_values_list, dim=0)
+            elif pixels.dim() != 4:
+                raise ValueError(f"pixel_values_videos of shape {pixels.shape}, expect to be of 4 or 5 dimensions")
```

Comment:
this is error a bit misleading for the user: 
* `pixels` here could he for `pixel_values_videos` or `pixel_values`
* I don't think `pixel_values_videos` should be of dim 4 or 5

Reply:
Yes, maybe we'll need to raise an error before flattening to check number of dims is correct